src/network/network_survey.cpp

```diff
@@ -0,0 +1,409 @@
+	{NetworkSurveyHandler::Reason::EXIT, "exit"},
+	{NetworkSurveyHandler::Reason::CRASH, "crash"},
+})
+
+NLOHMANN_JSON_SERIALIZE_ENUM(GRFStatus, {
+	{GRFStatus::GCS_UNKNOWN, "unknown"},
+	{GRFStatus::GCS_DISABLED, "disabled"},
+	{GRFStatus::GCS_NOT_FOUND, "not found"},
+	{GRFStatus::GCS_INITIALISED, "initialised"},
+	{GRFStatus::GCS_ACTIVATED, "activated"},
+})
+
+static const std::string _vehicle_type_to_string[] = {
+	"train",
+	"roadveh",
+	"ship",
+	"aircraft",
+};
+
+/* Defined in one of the os/ survey files. */
+extern void SurveyOS(nlohmann::json &json);
+
+/**
+ * List of all the generic setting tables.
+ *
+ * There are a few tables that are special and not processed like the rest:
+ * - _currency_settings
+ * - _misc_settings
+ * - _company_settings
+ * - _win32_settings
+ * As such, they are not part of this list.
+ */
+static auto &GenericSettingTables()
+{
+	static const SettingTable _generic_setting_tables[] = {
+		_difficulty_settings,
+		_economy_settings,
+		_game_settings,
+		_gui_settings,
+		_linkgraph_settings,
+		_locale_settings,
+		_multimedia_settings,
+		_network_settings,
+		_news_display_settings,
+		_pathfinding_settings,
+		_script_settings,
+		_world_settings,
+	};
+	return _generic_setting_tables;
+}
+
+/**
+ * Convert a settings table to JSON.
+ *
+ * @param survey The JSON object.
+ * @param table The settings table to convert.
+ * @param object The object to get the settings from.
+ */
+static void SurveySettingsTable(nlohmann::json &survey, const SettingTable &table, void *object)
+{
+	char buf[512];
+	for (auto &desc : table) {
+		const SettingDesc *sd = GetSettingDesc(desc);
+		/* Skip any old settings we no longer save/load. */
+		if (!SlIsObjectCurrentlyValid(sd->save.version_from, sd->save.version_to)) continue;
+
+		auto name = sd->GetName();
+		sd->FormatValue(buf, lastof(buf), object);
+
+		survey[name] = buf;
+	}
+}
+
+/**
+ * Convert settings to JSON.
+ *
+ * @param survey The JSON object.
+ */
+static void SurveySettings(nlohmann::json &survey)
+{
+	SurveySettingsTable(survey, _misc_settings, nullptr);
+#if defined(_WIN32) && !defined(DEDICATED)
+	SurveySettingsTable(survey, _win32_settings, nullptr);
+#endif
+	for (auto &table : GenericSettingTables()) {
+		SurveySettingsTable(survey, table, &_settings_game);
+	}
+	SurveySettingsTable(survey, _currency_settings, &_custom_currency);
+	SurveySettingsTable(survey, _company_settings, &_settings_client.company);
+}
+
+/**
+ * Convert generic OpenTTD information to JSON.
+ *
+ * @param survey The JSON object.
+ */
+static void SurveyOpenTTD(nlohmann::json &survey)
+{
+	survey["version"] = std::string(_openttd_revision);
+	survey["newgrf_version"] = _openttd_newgrf_version;
+	survey["build_date"] = std::string(_openttd_build_date);
+	survey["bits"] =
+#ifdef POINTER_IS_64BIT
+			64
+#else
+			32
+#endif
+		;
+	survey["endian"] =
+#if (TTD_ENDIAN == TTD_LITTLE_ENDIAN)
+			"little"
+#else
+			"big"
+#endif
+		;
+	survey["dedicated"] =
```

Comment:
How important is it whether the client was compiled as dedicated server? If you just install the standard OpenTTD, then this will be "no" and we'd still have to look to the video driver for the actual information.
If it is important, then maybe rename it to "dedicated_build", to make it clear that it's about the choice made at building the game not when starting the game.

Reply:
Personally, don't actually care. But it was in the crashlog .. so someone once thought it was important :P `dedicated_build` it is :)

(I think it is good to keep it, on second thought .. shows us a bit how often people make a dedicated server build, as in: is it worth supporting the option. I think it is, and this would make it statistical :D)